### PR TITLE
Extract session service

### PR DESCRIPTION
**Pull request by bllucia**
Extract session service to x-utils (and fixing ttl bug)

### CODE REVIEW

File: packages/x-utils/src/services/session.service.ts [hunk 1]
```diff
@@ -21,7 +25,9 @@ export class DefaultSessionService implements SessionService {
   public static instance: SessionService = new DefaultSessionService();
 
   public constructor(
-    protected storageService: StorageService = new BrowserStorageService(localStorage, 'x'),
+    protected storageService: StorageService = typeof localStorage !== 'undefined'
+      ? new BrowserStorageService(localStorage, 'x')
+      : new InMemoryStorageService(),
```
Comment (javieri-empathy) [inline packages/x-utils/src/services/session.service.ts:30]: I'd like to remove this singleton and the default parameters so it is not coupled to the browser. The service instantiation should be done somewhere else in x-components. Initially the idea we had with services was to add them to the x-module. But for this PR I think it is fine. We can work this in the future.

**Review by javieri-empathy — DISMISSED** [review]

File: packages/x-components/src/x-modules/tagging/wiring.ts [hunk 2]
```diff
@@ -8,8 +9,7 @@ import { wireService, wireServiceWithoutPayload } from '../../wiring/wires.facto
 import { filter, mapWire } from '../../wiring/wires.operators';
 import { Wire } from '../../wiring/wiring.types';
 import { createWiring } from '../../wiring/wiring.utils';
-import { DefaultSessionService } from './service/session.service';
-import { DefaultPDPAddToCartService } from './service/index';
+import { DefaultPDPAddToCartService } from './service';
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/tagging/wiring.ts:12]: Avoid importing from barrel file

```suggestion
import { DefaultPDPAddToCartService } from './service/pdp-add-to-cart.service';

```

**Review by tiborux — APPROVED** [review]

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
Good job!

**Review by tiborux — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**